src/main/java/edu/nitt/delta/bustracker/service/BusTrackerUserDetailsService.java

```diff
@@ -0,0 +1,36 @@
+package edu.nitt.delta.bustracker.service;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import edu.nitt.delta.bustracker.model.User;
+import edu.nitt.delta.bustracker.repository.UserRepository;
+
+@Service
+public class BusTrackerUserDetailsService implements UserDetailsService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String mobileNumber) throws UsernameNotFoundException {
+        User user = userRepository.findByMobileNumber(mobileNumber).orElse(null);
+
+        if (user == null) {
+            throw new UsernameNotFoundException("Mobile number not found.");
+        }
+
+        Collection<SimpleGrantedAuthority> authorities = new ArrayList<>();
+        authorities.add(new SimpleGrantedAuthority(user.getRole().toString()));
```

Comment:
Prefix it with "ROLE_" eg, "ROLE_DRIVER" or "ROLE_ADMIN" so that it's a "role" rather than an "authority".